<launch>
  <arg name="camera_id" default="0" /> <!-- 0 for main camera, 1 for tracking camera and 2 for webcames. -->
  <arg name="frame_rate" default="30" />
  <arg name="width" default="1920" /> <!-- 1920 for main camera, 1280 for tracking camera --> 
  <arg name="height" default="1080" /> <!-- 1080 for main camera, 720 for tracking camera -->
  <arg name="input_format" default="NV12" /> <!-- format for the input stream from camera -->
  <arg name="output_format" default="RGB" /> <!-- format for the output stream -->
  <arg name="topic_name_prefix" default="camera" />
  <arg name="image_compress" default="true" /> 

  <node pkg="rb5_vision" type="rb5_vision" name="rb5_vision" output="screen">
    <param name="camera_id" value="$(arg camera_id)" />
    <param name="frame_rate" value="$(arg frame_rate)" />
    <param name="width" value="$(arg width)" />
    <param name="height" value="$(arg height)" />
    <param name="input_format" value="$(arg input_format)" />
    <param name="output_format" value="$(arg output_format)" />
    <param name="topic_name" value="$(arg topic_name_prefix)_$(arg camera_id)" />
    <param name="image_compress" value="$(arg image_compress)" />
  </node>
</launch>
